lib/common/pacemakerd_client.c

```diff
@@ -0,0 +1,299 @@
+/*
+ * Copyright 2004-2019 the Pacemaker project contributors
+ *
+ * The version control history for this file may have further details.
+ *
+ * This source code is licensed under the GNU Lesser General Public License
+ * version 2.1 or later (LGPLv2.1+) WITHOUT ANY WARRANTY.
+ */
+
+#include <crm_internal.h>
+#include <stdlib.h>
+#include <time.h>
+
+#include <crm/msg_xml.h>
+#include <crm/common/mainloop.h>
+#include <crm/common/pacemakerd_types.h>
+
+CRM_TRACE_INIT_DATA(pacemakerd);
+
+typedef struct pacemakerd_api_private_s {
+    mainloop_io_t *source;
+    void (*ping_callback) (pacemakerd_t *pacemakerd, time_t last_good,
+                           enum pacemakerd_state state, int rc,
+                           gpointer userdata);
+    gpointer ping_userdata;
+    void (*disconnect_callback) (gpointer userdata);
+    gpointer disconnect_userdata;
+} pacemakerd_api_private_t;
+
+const char *pacemakerd_state_str[] = {
+    XML_PING_ATTR_PACEMAKERDSTATE_INIT,
+    XML_PING_ATTR_PACEMAKERDSTATE_STARTINGDAEMONS,
+    XML_PING_ATTR_PACEMAKERDSTATE_WAITPING,
+    XML_PING_ATTR_PACEMAKERDSTATE_RUNNING,
+    XML_PING_ATTR_PACEMAKERDSTATE_SHUTTINGDOWN,
+    XML_PING_ATTR_PACEMAKERDSTATE_SHUTDOWNCOMPLETE
```

Comment:
There are no XML_.*_ATTR_.* defines in any other header so far.
We could split msg_xml.h into multiple files and include them in msg_xml.h so that code using msg_xml.h wouldn't have to change. If we do the includes where the defines had previously
been we wouldn't even mess up the whole structure.
For this very case though we already have the enum and the converters in both directions.
So we could keep the strings entirely inside the c-file.

Reply:
These aren't XML element or attribute names, so I was thinking we could rename them when moving them. E.g. if we went with pcmk_core_ we would use PCMK_CORE_STATE_*.

Reply:
But right, in this case we can now make them private to the file